### PR TITLE
Fix: Symlink configuration files are not read as actual config files

### PR DESCRIPTION
**Pull request by agentx3**
### What does it do?

I identified where the config where being loaded, and modified the filter function.
The filter function now:
1) Checks if the file is a symlink
2) If it is a symlink, verify it's pointing to a valid file

Original handling of real files should not be affected.

Added tests for the change.

### Why is it needed?

There are many reasons why the config files might be symlinks. In my case, I'm building a docker image using nix, so all files are actually symlinks  to paths inside /nix/store. The linked issue https://github.com/strapi/strapi/issues/15667 also gives reasons.

### How to test it?
Replace the configuration files inside dist/config with symlinks (of the same name) to real config files. 

### Related issue(s)/PR(s)

https://github.com/strapi/strapi/issues/15667

### CODE REVIEW

File: packages/core/strapi/src/core/app-configuration/config-loader.ts [hunk 1]
```diff
@@ -9,7 +9,17 @@ export default (dir: string) => {
 
   return fs
     .readdirSync(dir, { withFileTypes: true })
-    .filter((file) => file.isFile() && VALID_EXTENSIONS.includes(path.extname(file.name)))
+    .filter((file) =>{
+      const fullPath = path.join(dir, file.name);
+      let stat: fs.Stats | fs.Dirent;
+      try {
+        stat = file.isSymbolicLink() ? fs.statSync(fullPath) : file;
+      } catch(err) {
```
Comment (innerdvations) [inline packages/core/strapi/src/core/app-configuration/config-loader.ts:17]:
```suggestion
      } catch (err) {
```
linting

File: packages/core/strapi/src/core/app-configuration/config-loader.ts [hunk 2]
```diff
@@ -9,7 +9,17 @@ export default (dir: string) => {
...
+        // ignore broken symbolic link
+        return false;
```
Comment (innerdvations) [inline packages/core/strapi/src/core/app-configuration/config-loader.ts:19]: We might want to still throw the error the same as if we attempt to load a .js file that has errors and can't be loaded.

Thoughts?

File: packages/core/strapi/src/core/app-configuration/config-loader.ts (hunk 1 above)
Comment (agentx3) [reply packages/core/strapi/src/core/app-configuration/config-loader.ts]: I've fixed linting issues, `npm run lint:other` on both files pass for me

File: packages/core/strapi/src/core/app-configuration/config-loader.ts (hunk 2 above)
Comment (agentx3) [reply packages/core/strapi/src/core/app-configuration/config-loader.ts]: Good idea, I was too tunnel visioned on keeping the original behavior 😅. I've added an error throw as well as adapted the tests for the the change.

**Comment by alexandrebodin** [comment]
> [!IMPORTANT] 
> With the release of Strapi 5, we have lifted the code freeze. To ensure we can effectively review and assist with contributions, we are now only accepting PRs targeting version 5 (the develop branch).<br/> 
> As your PR targets version 4, it has been automatically closed. However, we would greatly appreciate it if you could update your contribution to align with version 5 and resubmit!

**Pull request closed without merging**